File: android/src/main/java/com/jhssong/simple_android_notification/models/NotificationData.java
package com.jhssong.simple_android_notification.models;

import java.util.Map;

public class NotificationData {
    public String channelId;
    public CharSequence title;
    public String text;
    public String payload;

    public NotificationData(String channelId, CharSequence title, String text, String payload) {
        this.channelId = channelId;
        this.title = title;
        this.text = text;
        this.payload = payload;
    }

    public static NotificationData from(Map<String, Object> arguments) {
        String channelId = (String) arguments.get("channelId");
        CharSequence title = (String) arguments.get("title");
        String text = (String) arguments.get("text");
        String payload = (String) arguments.get("payload");
        return new NotificationData(channelId, title, text, payload);
    }
}

File: lib/models/notification_data.dart
class NotificationData {
  String channelId;
  String title;
  String text;
  String payload;

  NotificationData({
    required this.channelId,
    required this.title,
    required this.text,
    required this.payload,
  });

  static Map<String, dynamic> toJson(NotificationData data) {
    return {
      'channelId': data.channelId,
      'title': data.title,
      'text': data.text,
      'payload': data.payload,
    };
  }
}
